### PR TITLE
Benchmark for finding resources

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

Adds a benchmark case for finding resources. Right now it contains only simple cases but it can be expanded later.

## What is the current behavior?

dotTrace showing 50% time spent casting
image

Benchmark results on my machine:

|                  Method |        Mean |     Error |     StdDev | Allocated |
|------------------------ |------------:|----------:|-----------:|----------:|
|         FindPreResource | 1,716.78 us | 86.903 us | 256.235 us |       1 B |
|        FindPostResource |    42.62 us |  1.943 us |   5.668 us |         - |
| FindNotExistingResource | 1,580.46 us | 57.811 us | 170.456 us |       1 B |

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019854-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: tests/Avalonia.Benchmarks/Styling/ResourceBenchmarks.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            var preHost = new Style();
+            preHost.Resources.Add("preTheme", null);
+            
+            var postHost = new Style();
+            postHost.Resources.Add("postTheme", null);
+            
+            return new Styles
+            {
+                preHost,
+                new FluentTheme(new Uri("avares://Avalonia.Benchmarks")),
```
Comment (grokys) [inline tests/Avalonia.Benchmarks/Styling/ResourceBenchmarks.cs:50]: My only worry about this is that changes to the fluent theme could affect benchmarks, which is fine for benchmarking the fluent theme, but this benchmark should really be about resource lookups I think?

Comment (MarchingCube) [reply tests/Avalonia.Benchmarks/Styling/ResourceBenchmarks.cs:50]: We can create a placeholder theme. I was basically looking for a large set of styles/resources.

Comment (MarchingCube) [reply tests/Avalonia.Benchmarks/Styling/ResourceBenchmarks.cs]: Changed to generate styles instead. Got similar benchmark results.

|                  Method |        Mean |     Error |    StdDev | Allocated |
|------------------------ |------------:|----------:|----------:|----------:|
|         FindPreResource | 1,086.21 us | 22.644 us | 66.412 us |       1 B |
|        FindPostResource |    41.99 us |  0.838 us |  2.056 us |         - |
| FindNotExistingResource | 1,129.00 us | 22.471 us | 62.268 us |       1 B |

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019917-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019942-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**